src/dbnode/storage/namespace_readers.go

```diff
@@ -251,11 +255,14 @@ func (m *namespaceReaderManager) get(
 	// We have a closed reader from the cache (either a cached closed
 	// reader or newly allocated, either way need to prepare it)
 	reader := lookup.closedReader
+	// TODO(juchan): get the actual volume here.
```

Comment:
You'll have to make the volume index be part of the `readerPosition` too and make that part of the page token perhaps returned to clients.

Reply:
Yeah I think thats the right approach although actual implementation might be slightly more complicated:

```
1. Return `readPosition` that includes volume number
2. RPC comes in with `readPosition`. If volume still exists, continue reading. If volume does not exist, start from the beginning at the next volume
```

or something like that. The peers bootstrapper is already designed to handle duplicate metadata so this should be fine

Reply:
Tracking via https://github.com/m3db/m3/issues/1747.